EventListener/LeadSubscriber.php

```diff
@@ -31,14 +31,19 @@ class LeadSubscriber extends CommonSubscriber
     /** @var array */
     protected $aliases;
 
+    /** @var array */
+    protected $seen;
+
     /**
      * LeadSubscriber constructor.
      *
      * @param ExtendedFieldModel $leadModel
      */
     public function __construct(ExtendedFieldModel $leadModel)
     {
-        $this->leadModel = $leadModel;
+        $this->leadModel      = $leadModel;
+        $this->extendedFields = $leadModel->getExtendedFields();
```

Comment:
This means the extendedField array has to be built even before any events inside this class are fired (unless I am mistaken). This could be bad for global performance.

Reply:
reverted